PD-487: Update TrueNAS Security Best Practices (#1947)
truenas/documentation: content/Solutions/Optimizations/Security.md

---
title: "Security Recommendations"

> description: "Best practices to minimize security vulnerabilities in TrueNAS."

Reviewer: "Best practices to operate TrueNAS in a secure manner"

weight: 30
aliases:
  - /core/solutions/optimizations/security/
tags:
 - coressh
 - scalessh
keywords:
 - TrueNAS Security
---

{{< toc >}}

> Follow these best practices to minimize security vulnerabilities in TrueNAS.

Reviewer: "Minimize security vulnerabilities" doesn't sound great. Should focus on the positive angle, about best practices to operate TrueNAS in a secure manner.

## General Recommendations

> 

Reviewer 2: Is this a step procedure or a list of recommendations? Parallel construction is off on a few points.

> * Specially modifying the base TrueNAS firmware image is unsupported and can create security issues.

Reviewer 2: Parallel construction issue, maybe this?
```suggestion
* Modifying  the base TrueNAS firmware image is unsupported and can create security issues.
```

> 2. Keep TrueNAS up to date with the most recent maintenance releases of the installed major version.

Reviewer: "Keep TrueNAS up to date with the most recently updates for your supported version"

3. Upgrade to new major releases in a timely manner consistent with the deployment use case.
4. Restrict new TrueNAS user accounts to the most minimal set of permissions and access possible.

> 5. Never grant TrueNAS user accounts (local or Directory Services added accounts) access to SSH or login Shells unless that account is explicitly trusted.

Reviewer: "to SSH or console shells"

> * Never grant TrueNAS user accounts (local or Directory Services added accounts) access to SSH or console shells unless that account is explicitly trusted.

Reviewer 2: Parallel construction issue, maybe this?
```suggestion
* Grant TrueNAS user accounts (local or Directory Services added accounts) access to SSH or console shells only if  that account is explicitly trusted.
```

7. Disable any Network services not in use.
8. Restrict the TrueNAS web, IPMI, and any other management interfaces to private subnets away from untrusted users.
9. Review any plugin, App, or Virtual Machine (VM) deployment scenario for additional security exposure or vulnerabilities.
    iXsystems cannot resolve security vulnerabilities introduced from within user-deployed virtualized environments.

## Hardening Specific System Services

### iSCSI

Follow the iSCSI creation wizard unless a specific configuration is required.

> To create an iSCSI share, go to **Sharing > Block Shares (iSCSI)** and click *WIZARD*.

Reviewer 2: line 41 the option to add should be in bold not italics.
Line 44 Initiators and Authorized Networks should be bold not italics.

This allows defining which systems or networks can connect to the extent.
When these options are empty, all initiators and all networks are allowed to connect to the extent.

### NFS

Network File System (NFS) is a sharing protocol that allows outside users to connect and view or modify shared data.

> 

Reviewer 2: Line 59 Authorized Hosts and IP addresses should be bold not italics.

To define which systems can connect to the share, click the **Advanced Options** and enter all networks, hosts, and IP addresses to have share access.
All other systems are denied access.

### SMB

Using Server Message Block (SMB) to share data is a very common situation for TrueNAS users.

> However, it allows outside connections to the system and must be properly use to avoid security concerns.

Reviewer 2: incorrect use of italics in line 74, 77, 80, 83, and 85, all the UI elements should be bold
